File: backend/app/schemas/media.py
from datetime import datetime

from pydantic import BaseModel, Field


class MediaReference(BaseModel):
    source: str = Field(..., min_length=1)
    label: str = Field(..., min_length=1)


class MediaFileItem(BaseModel):
    filename: str
    relative_path: str
    url: str
    content_type: str
    media_type: str
    size: int
    modified_at: datetime
    referenced: bool
    references: list[MediaReference] = Field(default_factory=list)


class MediaListResponse(BaseModel):
    items: list[MediaFileItem] = Field(default_factory=list)
    total: int
    used_count: int
    unused_count: int
    total_size: int
    unused_size: int


class MediaCleanupResponse(BaseModel):
    deleted_count: int
    deleted_size: int
    deleted_files: list[str] = Field(default_factory=list)

File: admin/src/types/media.ts
export type MediaReference = {
  source: string;
  label: string;
};

export type MediaFileItem = {
  filename: string;
  relative_path: string;
  url: string;
  content_type: string;
  media_type: "image" | "resume" | "other" | string;
  size: number;
  modified_at: string;
  referenced: boolean;
  references: MediaReference[];
};

export type MediaListResponse = {
  items: MediaFileItem[];
  total: number;
  used_count: number;
  unused_count: number;
  total_size: number;
  unused_size: number;
};

export type MediaCleanupResponse = {
  deleted_count: number;
  deleted_size: number;
  deleted_files: string[];
};
